clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -13,6 +13,7 @@
 #include "clang/CIR/Dialect/IR/CIRDialect.h"
 #include "clang/AST/Attrs.inc"
 #include "clang/CIR/Dialect/IR/CIRAttrs.h"
+#include "clang/CIR/Dialect/IR/CIRDataLayout.h"
```

Comment:
Sorry for the horrible changes. I have to include `::cir::CIRDataLayout` here. CE because `::mlir::cir` and `::cir` are ambiguous.

Reply:
Not sure I get this, if you start with `::` it's usually enough to remove the ambiguity, so referring to `::cir::CIRDataLayout` should be enough, while keeping the rest as is. Also, for things like this, where unrelated changes are a bit massive, they should be prep PRs - I still think you don't need these in this specific case though. What did you tried and what didn't work?

Reply:
Indeed I can specify `::cir` for `CIRDataLayout`. But then it's original codes like `void cir::CIRDialect::initialize()` being ambiguous and causes a bunch of compilation errors.

I think it's because C++ lookups names hierarchically and failed in the unqualified name `cir` already without consider there's actually nothing called `::cir::CIRDialect`. You may try solving [this minimal repro](https://godbolt.org/z/zExM6x41x) individually ; )

I also tried to come up with an approach that does not touch all usage of `cir`. I can also implement `AllocaOp::verify()` in another `.cpp` file (even a new one). But the current changes has the least disadvantages (if have).

Reply:
The idea I'm trying to convey is that we should do the best for a specific file, instead of workarounds that don't take the big picture into consideration. CIRDialect.cpp is mainly about implementing stuff under `mlir::cir`, therefore that namespace should be the one taking priority in this file.

Changed you repro to this: https://godbolt.org/z/zfnE6o67r, like I suggested above, and it works for me. So not really sure what you are talking about. It's possible that if even so we get into problems, might be worth deleting the `using namespace mlir;` which is a bit aggressive, but I think we can workaround this restriction in the way described in https://godbolt.org/z/zfnE6o67r

Reply:
> Indeed I can specify `::cir` for `CIRDataLayout`. But then it's original codes like `void cir::CIRDialect::initialize()` being ambiguous that cause a bunch of compilation errors.

☝️ , there are already many `cir::`-qualified symbols in the TU (mixed with `::mlir::cir` and `mlir::cir`), so just including the data layout header breaks it due to the ambiguities. I played around a little but also didn't find a solution that works without re-qualifying at least some of the symbols in the file.

Reply:
I feel like `using namespace mlir;` is a bit too much for this file. Looks like we'll need some rewriting anyways, but I'd argue in favor of keeping `using namespace mlir::cir;` and fixing the pure `::mlir` ones. I'm in favor of such change, but not as part of this PR, should be its own PR.

My impression though is that by removing the verification bits we probably won't need to look at `CIRDataLayout` anymore in `CIRDialect.cpp`.